packages/embark-ui/src/services/storage.js

```diff
@@ -2,6 +2,14 @@
 
 export async function addEditorTabs({file}) {
   const editorTabs = findOrCreateEditorTabs();
+
+  // Avoid files bigger than 1MB. Browsers limit heavily
+  // how much local storage we can actually use.
+  if(file.content.length > 1024 * 1024) {
+    alert('File is too big');
```

Comment:
This is most likely not the right place for something along these lines, but it reflects the intent. I'm open to ideas.